embulk-core/src/main/java/org/embulk/cli/CommandLine.java

```diff
@@ -0,0 +1,18 @@
+package org.embulk.cli;
+
+import java.util.List;
+import java.util.Properties;
+
+public abstract class CommandLine {
```

Comment:
curious, why not this one an interface instead of abstract class?

Reply:
No strong reasons. 
* It had some implemented methods when I started tries-and-errors on this, but they finally got away.
* This is just a stub of `embulk-deps`' `CommandLineImpl`, not expecting to have various classes implementing the `CommandLine` interface.